pkg/process/config/yaml_config.go

```diff
@@ -216,15 +218,31 @@ func (a *AgentConfig) LoadProcessYamlConfig(path string) error {
 	// use `internal_profiling.enabled` field in `process_config` section to enable/disable profiling for process-agent,
 	// but use the configuration from main agent to fill the settings
 	if config.Datadog.IsSet(key(ns, "internal_profiling.enabled")) {
-		a.ProfilingEnabled = config.Datadog.GetBool(key(ns, "internal_profiling.enabled"))
-		a.ProfilingSite = config.Datadog.GetString("site")
-		a.ProfilingURL = config.Datadog.GetString("internal_profiling.profile_dd_url")
-		a.ProfilingEnvironment = config.Datadog.GetString("env")
-		a.ProfilingPeriod = config.Datadog.GetDuration("internal_profiling.period")
-		a.ProfilingCPUDuration = config.Datadog.GetDuration("internal_profiling.cpu_duration")
-		a.ProfilingMutexFraction = config.Datadog.GetInt("internal_profiling.mutex_profile_fraction")
-		a.ProfilingBlockRate = config.Datadog.GetInt("internal_profiling.block_profile_rate")
-		a.ProfilingWithGoroutines = config.Datadog.GetBool("internal_profiling.enable_goroutine_stacktraces")
+		// allow full url override for development use
+		s := config.DefaultSite
+		cfgSite := config.Datadog.GetString("site")
+		cfgURL := config.Datadog.GetString("internal_profiling.profile_dd_url")
+		if cfgSite != "" {
+			s = cfgSite
+		}
+
+		site := fmt.Sprintf(profiling.ProfileURLTemplate, s)
+		if cfgURL != "" {
+			site = cfgURL
+		}
```

Comment:
I think the logic could be simplified like so, wdyt?

```suggestion
		site := config.Datadog.GetString("internal_profiling.profile_dd_url")
		if site == "" {
			s := config.Datadog.GetString("site")
			if s == "" {
				s = config.DefaultSite
			}
			site = fmt.Sprintf(profiling.ProfileURLTemplate, s)
		}
```